DTS-STN/canadian-dental-care-plan
# Mapping `applicationYear.id` to Interop request payload

### Description
- For the renewal route helpers, all the `validateStateForReview` functions need to accept and return the `applicationYear` object.
- Added a new `applicationYearId` field to the `BenefitRenewalDto`. The state mapper maps the `applicationYear` object in state to this new field.
- The DTO mapper maps the `applicationYearId` field to the Interop request payload (`BenefitApplicationYear` field) to submit a renewal application.

### Related Azure Boards Work Items
AB#4860

### Screenshots (if applicable)
![image](https://github.com/user-attachments/assets/9e403e34-a2ae-4c49-bbeb-df03f617438b)

### Checklist

### Test Instructions
Run application and go through the renewal flow until you reach the review page and verify that the request payload has the `BenefitApplicationYear` populated (feature to view request payload and Power Platform mock must both be enabled)
